stages/org.osbuild.ostree.deploy

```diff
@@ -104,44 +131,93 @@ def make_fs_identifier(desc):
 
 
-def main(tree, inputs, options):
-    osname = options["osname"]
-    rootfs = options.get("rootfs")
-    mounts = options.get("mounts", [])
-    kopts = options.get("kernel_opts", [])
-    ref = options["ref"]
-    remote = options.get("remote")
+def ostree_commit_deploy(tree, inputs, osname, remote, ref, kopts):
+    if len(inputs) == 0:
+        if not ref:
+            raise ValueError("ref should be specified in options")
+    elif len(inputs) == 1:
+        if ref:
+            raise ValueError("Should not specify ref if input was specified")
 
-    # If provided an input then do the pull into the tree
-    if len(inputs) != 0:
+        # If we have an input then we need to pull_local() from the input
+        # first before we deploy.
         source_repo, commits = ostree.parse_input_commits(inputs["commits"])
         target_repo = f"{tree}/ostree/repo"
         for commit, data in commits.items():
-            loopref = data.get("ref", commit)
-            ostree.pull_local(source_repo, target_repo, remote, loopref)
+            ref = data.get("ref", commit)
+            ostree.pull_local(source_repo, target_repo, remote, ref)
 
     if remote:
         ref = f"{remote}:{ref}"
 
-    kargs = []
+    kargs = [f'--karg-append={v}' for v in kopts]
+    ostree.cli("admin", "deploy", ref,
+               *kargs, sysroot=tree, os=osname)
+
+
+def ostree_container_deploy(tree, inputs, osname, remote, target_gpgcheck, kopts):
+    images = containers.parse_containers_input(inputs)
+    for image in images.values():
+        with containers.container_source(image) as (image_name, image_source):
+            extra_args = []
+            imgref = f"ostree-unverified-image:{image_source}"
+
+            if target_gpgcheck:
+                target_imgref = 'ostree-remote-registry'
+            else:
+                target_imgref = 'ostree-unverified-registry'
+
+            if remote:
+                # If the user specified an ostree remote it's most likely
+                # because they want to take advantage of GPG signature
+                # verification of the OSTree commits inside the image.
+                # Add it to the imgref spec here.
+                target_imgref += f':{remote}'
+            target_imgref += f':{image_name}'
+
+            extra_args.append(f'--imgref={imgref}')
+            extra_args.append(f'--stateroot={osname}')
+            extra_args.append(f'--target-imgref={target_imgref}')
+
+            kargs = [f'--karg={v}' for v in kopts]
+
+            ostree.cli("container", "image", "deploy",
+                       *extra_args, sysroot=tree, *kargs)
+
+
+def main(tree, inputs, options):
+    osname = options["osname"]
+    rootfs = options.get("rootfs")
+    mounts = options.get("mounts", [])
+    kopts = options.get("kernel_opts", [])
+    ref = options.get("ref", "")
+    remote = options.get("remote")
+    target_gpgcheck = options.get("target_container_signature_verification", True)
```

Comment:
Mmm...this isn't about GPG here but about the skopeo polcy.  I'd just call this `target_signature_verification` - seems simpler.

Reply:
That works for me.